cryptodoc/src/01_hash.rst

```diff
@@ -78,3 +78,16 @@ data compression.
 
 .. [#sha_intrinsics]
    https://github.com/noloader/SHA-Intrinsics
+
+BLAKE2b
+-------
+
+Botan implements the hash function BLAKE2b as defined in [RFC7693]_.
+The implementation is located in :srcref:`src/lib/hash/blake2/blake2b.cpp`.
+BLAKE2s is not supported by Botan.
+
+As defined in [RFC7693]_ Botan's BLAKE2b implementation allows for an arbitrary
+number of up to 64 output bytes with the respective security implications.
+Also, it contains an interface to initialize the hash function with a secret
+key. As described in [RFC7693]_, the key can have an arbitrary size of up to 64
```

Comment:
On that note: maybe mention BLAKE2 in the Message Authentication Code section of this document and simply link to this section. Along those lines:

```rst
BLAKE2b
-------

The BLAKE2b construction allows for an optional ``key`` parameter of up to 64 bytes length.
It can therefore be used as a message authentication code.
In particular BLAKE2b does not require to be used within an HMAC construction.
For further details see: :ref:`hash/blake2b`. 

Botan provides a thin wrapper of BLAKE2b in :srcref:`src/lib/mac/blake2mac/blake2bmac.h`
to make it available as a subclass of ``Botan::MessageAuthenticationCode``.
```

Don't forget adding an rST label above the BLAKE2b headline of this chapter:

```rst
.. _hash/blake2b:
```

Reply:
Isn't `blake2bmac` a separate module that is out of scope?

Reply:
Mhh, true that. Agreed. Blake2b is just pulled in as a dependencies of Argon2.